src/main/java/seedu/address/logic/commands/MatchAutoCommand.java

```diff
@@ -0,0 +1,138 @@
+    private List<Buyer> buyers; // Buyers to match
+    private Set<Property> matchedProperties; // Properties that have been matched
+    private Set<Buyer> matchedBuyers; // Buyers that have been matched
+    private List<Match> matches; // Confirmed matches
+
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        initialise(model);
+        runMatching();
+        sortMatches();
+        updateModel(model);
+        return getCommandResult();
+    }
+
+    private void initialise(Model model) throws CommandException {
+        if (model.getFilteredPropertyList().isEmpty()) {
+            throw new CommandException(MESSAGE_NO_PROPERTIES_LISTED);
+        }
+        if (model.getFilteredBuyerList().isEmpty()) {
+            throw new CommandException(MESSAGE_NO_BUYERS_LISTED);
+        }
+        properties = new ArrayList<>(model.getFilteredPropertyList());
+        buyers = new ArrayList<>(model.getFilteredBuyerList());
+        matchedProperties = new HashSet<>();
+        matchedBuyers = new HashSet<>();
+        matches = new ArrayList<>();
+    }
+
+    /**
+     * Runs matching algorithm to find best matches.
+     */
+    private void runMatching() {
+        List<Match> sortedMatchCandidates = getSortedMatchCandidates(); // Get all possible matches
+        for (Match match : sortedMatchCandidates) {
+            if (matchIsAvailable(match)) { // If buyer and property not taken
+                confirmMatch(match);
+            }
+        }
+    }
+
+    /**
+     * Sorts matches by decreasing desirability (match score and price difference).
+     */
+    private void sortMatches() {
+        // Comparator comparing budget - price
+        Comparator<Match> priceDiffComparator = Comparator.comparing(
+            m -> m.getBuyer().getMaxPrice().value - m.getProperty().getPrice().value);
+
+        // Comparing by increasing match score then budget - price
+        Comparator<Match> desirabilityComparator =
+                Comparator.comparing(Match::getMatchScore).thenComparing(priceDiffComparator);
+
+        matches.sort(desirabilityComparator.reversed());
+    }
+
+    private void updateModel(Model model) {
+        model.setMatchList(matches);
+    }
+
+    private CommandResult getCommandResult() {
+        String feedback;
+        if (matches.size() == 1) {
+            feedback = MESSAGE_ONE_MATCH_FOUND;
+        } else {
+            feedback = String.format(MESSAGE_MATCHES_FOUND, matches.size());
+        }
+        return new CommandResult(feedback, UiAction.SHOW_MATCHES);
+    }
+
+    /**
+     * Returns possible matches in descending order of desirability.
+     */
+    private List<Match> getSortedMatchCandidates() {
+        buyers.sort(Buyer.getPriceComparator()); // Prefer matching low budget buyers first
+        properties.sort(Property.getPriceComparator().reversed()); // Prefer matching expensive properties first
```

Comment:
this can lead to the possibility of a person paying for a property way above the budget, although the tags mostly match
Some thinking is required for the matching criteria. Should we allow for matches that exceed the budget?

Reply:
money can't buy happiness, but it can buy a dream property, and a dream property can give you happiness :)

The exact criteria can be tweaked, but allowing matches that exceed budget produces more matches and more information > less information right? :P